File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03940/p03940_num7_parsed.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int N, E, T;
    cin >> N >> E >> T;

    vector<int> bears(N);
    for (int i = 0; i < N; i++) {
        cin >> bears[i];
    }

    // Find the maximum time taken to reach the exit
    int maxTime = max(E - bears[N-1], bears[0]);

    // Calculate the minimum time to pick up all the coins and go to the exit
    int minTime = 0;
    for (int i = 1; i < N; i++) {
        int distance = bears[i] - bears[i-1];
        if (distance <= 2 * T) {
            minTime += distance;
        } else {
            minTime += 2 * T;
        }
    }
    minTime += bears[N-1] + T;

    // Output the answer
    cout << minTime + maxTime << endl;

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03940/p03940_num7_parsed.py
import sys

def minimum_time(N, E, T, positions):
    # Calculate the minimum time needed to collect all the coins and go to the exit
    min_time = 0
    
    # Calculate the time needed to give candies to all the bears
    for i in range(N):
        min_time += T * (E - positions[i])
    
    # Add the time needed to reach the exit
    min_time += E
    
    return min_time

# Read the input from standard input
input_lines = sys.stdin.read().strip().split("\n")
N, E, T = map(int, input_lines[0].split())
positions = list(map(int, input_lines[1].split()))

# Calculate the minimum time
result = minimum_time(N, E, T, positions)

# Print the result
print(result)